osmosis-apidb/src/main/java/org/openstreetmap/osmosis/apidb/v0_6/ApidbWriter.java

```diff
@@ -54,7 +54,8 @@
 	private static final int INSERT_PRM_COUNT_NODE = 8;
 
     private static final String INSERT_SQL_NODE_TAG_COLUMNS = "INSERT INTO node_tags (node_id, k, v, version)";
-    private static final String INSERT_SQL_NODE_TAG_PARAMS = "?, ?, ?, ?";
+    private static final String QUESTION_MARKS = "?, ?, ?, ?";
+    private static final String INSERT_SQL_NODE_TAG_PARAMS = QUESTION_MARKS;
```

Comment:
This makes the code very difficult to read.  A layer of indirection here serves little useful purpose.  The point of deduplicating strings is so that you don't have to update multiple locations when making a code change.  In this case each string performs a unique and specific purpose so centralising them just makes maintenance harder.  The name "QUESTION_MARKS" is way too generic.  I'd prefer to see a call to a function such as "createQuestionMarks(4)" or similar if we really want to eliminate duplicated strings and sonar warnings.